tools/licenses/lib/main.dart

```diff
@@ -462,40 +461,40 @@ class _RepositoryLibJpegTurboLicense extends _RepositoryLicenseFile {
     }
   }
 
-  List<License> _licenses;
+  List<License>? _licenses;
 
   @override
-  List<License> get licenses {
-    if (_licenses == null) {
-      final _RepositoryReadmeIjgFile readme = parent.getChildByName('README.ijg') as _RepositoryReadmeIjgFile;
-      final _RepositorySourceFile main = parent.getChildByName('turbojpeg.c') as _RepositorySourceFile;
-      final _RepositoryDirectory simd = parent.getChildByName('simd') as _RepositoryDirectory;
+  List<License>? get licenses {
+    if (_licenses == null && parent != null) {
```

Comment:
the `&& parent != null` should not be necessary

Reply:
Done.  I'm a bit puzzled why that works given `parent` is nullable. I guess there is an invariant that `get` isn't called when `parent` is null?

Reply:
yeah, the root case is special in various ways such that the code never gets called when parent is null.

This code was designed for a nullable world. I would write it quite differently now with null safety.